cassandra-release/cassandra-check-release.sh

```diff
@@ -0,0 +1,349 @@
+#!/bin/bash
+
+# Parameters
+# $1 staged|released
+# $2 release
+# $3 maven artefacts staging repo id (as specified in the repo url found in the vote email) (ignored for 'released')
+#
+# Example use: `./cassandra-check-release.sh staged 4.0-beta3 1224
+#
+# This script is very basic and experimental. I beg of you to help improve it.
+#
+
+###################
+# prerequisites
+
+command -v wget >/dev/null 2>&1 || { echo >&2 "wget needs to be installed"; exit 1; }
+command -v gpg >/dev/null 2>&1 || { echo >&2 "gpg needs to be installed"; exit 1; }
+command -v sha1sum >/dev/null 2>&1 || { echo >&2 "sha1sum needs to be installed"; exit 1; }
+command -v md5sum >/dev/null 2>&1 || { echo >&2 "md5sum needs to be installed"; exit 1; }
+command -v sha256sum >/dev/null 2>&1 || { echo >&2 "sha256sum needs to be installed"; exit 1; }
+command -v sha512sum >/dev/null 2>&1 || { echo >&2 "sha512sum needs to be installed"; exit 1; }
+command -v tar >/dev/null 2>&1 || { echo >&2 "tar needs to be installed"; exit 1; }
+command -v docker >/dev/null 2>&1 || { echo >&2 "docker needs to be installed"; exit 1; }
+(docker info >/dev/null 2>&1) || { echo >&2 "docker needs to running"; exit 1; }
+
+( [ "staged" = $1 ] || [ "released" = $1 ] ) || { echo >&2 "first argument must be staged or released"; exit 1; }
+( [ $# -ge 2 ] ) || { echo >&2 "minimum two arguments must be provided"; exit 1; }
+if [ -z "$3" ] ; then
+    [ "released" == $1 ] || { echo >&2 "third argument must not be specified when first is released"; exit 1; }
+    dist_url="https://dist.apache.org/repos/dist/release/cassandra/$2/"
+    maven_repo_url="https://repository.apache.org/content/repositories/releases/org/apache/cassandra/cassandra-all/$2"
+    debian_url="https://debian.cassandra.apache.org/"
+    redhat_url="https://redhat.cassandra.apache.org/"
+else
+    [ "staged" == $1 ] || { echo >&2 "third argument must be specified when first is staged"; exit 1; }
+    dist_url="https://dist.apache.org/repos/dist/dev/cassandra/$2/"
+    maven_repo_url="https://repository.apache.org/content/repositories/orgapachecassandra-$3/org/apache/cassandra/cassandra-all/$2"
+    debian_url="https://dist.apache.org/repos/dist/dev/cassandra/${2}/debian/"
+    redhat_url="https://dist.apache.org/repos/dist/dev/cassandra/${2}/redhat/"
+fi
+(curl --output /dev/null --silent --head --fail "${dist_url}") || { echo >&2 "Not Found: ${dist_url}"; exit 1; }
+(curl --output /dev/null --silent --head --fail "${maven_repo_url}") || { echo >&2 "Not found: ${maven_repo_url}"; exit 1; }
+
+###################
+
+idx=`expr index "$2" -`
```

Comment:
FYI osx has bsd expr which does not support the index command.

Reply:
i don't think this is the only problem  in the script with it running on osx.

i think solving osx support would be better done by moving more into docker runs, as @Mmuzaf points out here: https://github.com/apache/cassandra-builds/pull/32/files/fa4b258da36445cd9afd92959a2637bac516c2d7#r1272464191